File: 2021-09/모의고사.js
function solution(answers) {
    var answer = [];
    const ONE = [1, 2, 3, 4, 5];
    const TWO = [2, 1, 2, 3, 2, 4, 2, 5];
    const THREE = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    const [ONE_SIZE, TWO_SIZE, THREE_SIZE] = [ONE.length, TWO.length, THREE.length];
    
    let oneCount = 0;
    let twoCount = 0;
    let threeCount = 0;
    
    answers.forEach((num, idx) => {
        if(num === ONE[idx % ONE_SIZE]) oneCount++;
        if(num === TWO[idx % TWO_SIZE]) twoCount++;
        if(num === THREE[idx % THREE_SIZE]) threeCount++;
    });
    
    const MAX = Math.max(oneCount, twoCount, threeCount);
    if(oneCount === MAX) answer.push(1);
    if(twoCount === MAX) answer.push(2);
    if(threeCount === MAX) answer.push(3);
    
    return answer;
}

File: programmers/src/main/java/level1/모의고사.java
package level1;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class 모의고사 {

	int[] one = {1, 2, 3, 4, 5};
	int[] two = {2, 1, 2, 3, 2, 4, 2, 5, 2, 1, 2, 3, 2, 4, 2, 5};
	int[] three = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5, 3, 3, 1, 1, 2, 2, 4, 4, 5, 5};

	public int[] solution(int[] answers) {
		int oneSize = one.length;
		int twoSize = two.length;
		int threeSize = three.length;
		int answersSize = answers.length;

		int[] answer = {0, 0, 0};

		for (int i = 0; i < answersSize; i++) {
			if (answers[i] == one[i % oneSize]) {
				answer[0]++;
			}

			if (answers[i] == two[i % twoSize]) {
				answer[1]++;
			}

			if (answers[i] == three[i % threeSize]) {
				answer[2]++;
			}
		}

		int max = Arrays.stream(answer).max().getAsInt();

		List<Integer> result = new ArrayList<>();
		for (int i = 0; i < answer.length; i++) {
			if (answer[i] == max) {
				result.add(i + 1);
			}
		}

		return result.stream().mapToInt(Integer::intValue).toArray();
	}

	public static void main(String[] args) {
		모의고사 sol = new 모의고사();
		sol.solution(new int[]{1,2,3,4,5});
	}

}
